update all upload-artifact usage to v4

kdt3rd opened the pull request:
This replicates the fix Cary did for the main build to all other use locations

kdt3rd committed:
update all upload-artifact usage to v4

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
replicate unique name change

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd commented:
This addresses the rest of #1967 for the oss-fuzz and python CI components - @cary-ilm can you check I made the target names unique in the correct way? I just pattern matched what you had done the other day

cary-ilm commented on .github/workflows/python-wheels-publish.yml:
```diff
@@ -57,9 +57,9 @@ jobs:
+        uses: actions/upload-artifact@v4
         with:
-          name: wheels-${{ matrix.os }}
+          name: wheels-${{ matrix.os }}-${{ env.INSTALL_MANIFEST }}
```
Same thing here.
```suggestion
          name: wheels-${{ matrix.os }}
```

cary-ilm commented on .github/workflows/python-wheels-publish-test.yml:
```diff
@@ -64,9 +64,9 @@ jobs:
+        uses: actions/upload-artifact@v4
         with:
-          name: wheels-${{ matrix.os }}
+          name: wheels-${{ matrix.os }}-${{ env.INSTALL_MANIFEST }}
```
`-${{ env.INSTALL_MANIFEST }}` should definitely not get appended to the name, that was appropriate only for the install manifests in `ci_steps.yml`.

Leave this line as it was for now. This workflow gets run only during the release process so it's a little hard to test, but I'll review it before the next release to make sure it does the right thing.
```suggestion
          name: wheels-${{ matrix.os }}
```

cary-ilm committed:
Update .github/workflows/python-wheels-publish-test.yml

-${{ env.INSTALL_MANIFEST }} should definitely not get appended to the name, that was appropriate only for the install manifests in ci_steps.yml.

Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm committed:
Update .github/workflows/python-wheels-publish.yml

-${{ env.INSTALL_MANIFEST }} should definitely not get appended to the name, that was appropriate only for the install manifests in ci_steps.yml.

Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm commented on .github/workflows/ossfuzz_workflow.yml:
```diff
@@ -48,8 +48,8 @@ jobs:
       if: failure() && steps.build.outcome == 'success'
       with:
-        name: artifacts
+        name: artifacts-${{ env.INSTALL_MANIFEST }}
```
```suggestion
        name: artifacts
```

cary-ilm committed:
Update .github/workflows/ossfuzz_workflow.yml

Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm commented on .github/workflows/python-wheels.yml:
```diff
@@ -70,9 +70,9 @@ jobs:
+        uses: actions/upload-artifact@v4
         with:
-          name: wheels-${{ matrix.os }}
+          name: wheels-${{ matrix.os }}-${{ env.INSTALL_MANIFEST }}
```
```suggestion
          name: wheels-${{ matrix.os }}
```

cary-ilm committed:
Update .github/workflows/python-wheels.yml

Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm commented on .github/workflows/scorecard.yml:
```diff
@@ -43,9 +43,9 @@ jobs:
+        uses: actions/upload-artifact@v4
         with:
-          name: SARIF file
+          name: SARIF.file-${{ env.INSTALL_MANIFEST }}
```
```suggestion
          name: SARIF.file
```

cary-ilm committed:
Update .github/workflows/scorecard.yml

Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm commented on .github/workflows/scorecard.yml:
```diff
@@ -43,9 +43,9 @@ jobs:
+        uses: actions/upload-artifact@v4
         with:
-          name: SARIF file
+          name: SARIF.file
```
```suggestion
          name: SARIF file
```

cary-ilm committed:
Update .github/workflows/scorecard.yml

Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm commented:
@kdt3rd, if I did this right, I added commits that fix the artifact naming, where you'd copied `-${{ env.INSTALL_MANIFEST }}` where it's not necessary. I'm going to merge as is and we can fix it afterwards if necessary.
